controller/src/beerocks/master/son_management.cpp

```diff
@@ -1917,7 +1917,10 @@ void son_management::handle_bml_message(Socket *sd,
             op_error_code == eChannelScanOpErrCode::CHANNEL_SCAN_OP_SUCCESS) {
             auto channel_pool_set =
                 std::unordered_set<uint8_t>(channel_pool, channel_pool + channel_pool_size);
-            op_error_code = database.set_channel_scan_pool(radio_mac, channel_pool_set, false)
+            auto scan_all_channels =
```

Comment:
So ugly...

Reply:
Updated the code, and moved the logic to a helper function in the son_management